modules/data_release/ajax/AddPermission.php

```diff
@@ -12,25 +12,51 @@
  *  @link     https://github.com/aces/Loris
  */
 
-$DB =& Database::singleton();
-
+$DB = \Database::singleton();
 if ($_POST['action'] == 'addpermission') {
     $userid          = $_POST['userid'];
     $data_release_id = $_POST['data_release_id'];
-    $success         = $DB->insert(
-        'data_release_permissions',
-        array(
-         'userid'          => $userid,
-         'data_release_id' => $data_release_id,
-        )
-    );
 
-    $factory  = NDB_Factory::singleton();
-    $settings = $factory->settings();
+        $result = $DB->pselectOne(
+            "SELECT COUNT(*) FROM data_release_permissions
+             WHERE userid = :userid and
+             data_release_id = :data_release_id",
+            array(
+             'userid'          => $data_release_id,
+             'data_release_id' => $data_release_id,
+            )
+        );
+
+        $factory  = \NDB_Factory::singleton();
+        $settings = $factory->settings();
+        $baseURL  = $settings->getBaseURL();
+        if ($result != '1') {
+            $success  = $DB->insert(
+                'data_release_permissions',
+                array(
+                 'userid'          => $userid,
+                 'data_release_id' => $data_release_id,
+                )
+            );
+            $factory  = NDB_Factory::singleton();
+            $settings = $factory->settings();
```

Comment:
do these need to be called again if it's already done above?

Reply:
Nope! They should still be in scope